coalib/testing/LocalBearTestHelper.py

```diff
@@ -185,10 +201,11 @@ def check_results(self,
                                   create_tempfile=create_tempfile,
                                   tempfile_kwargs=tempfile_kwargs,
                                   settings=settings)
+        self.addTypeEqualityFunc(list, self.assertObjectsEqual)
```

Comment:
But we're basically comparing list of `Result` objects, so whenever we pass lists to the `assertEqual(...)` for comparison, it will call the `assertObjectsEqual(...)` method IIUC. What I had initially done was put `Result` object instead of `list` since we wanna call that method whenever we're comparing `Result` objects, see below : 
Snippet: `self.addTypeEqualityFunc(list, self.assertObjectsEqual)`
Output: 
```
E   AssertionError: 'mixed spaces and tabs' != 'mixed spaces and tab'
E   - mixed spaces and tabs
E   ?                     -
E   + mixed spaces and tab
E    : message_base mismatch
```
Snippet: `self.addTypeEqualityFunc(Result, self.assertObjectsEqual)`
Output: 
```
E   AssertionError: Lists differ: [<Result object(id=0xf33dc5cf898c48b4b7114cffd86e8172, origin='[490 chars]438>] != [<Result object(id=0xf211aef4abb24ecb8cc36beebbd848a8, origin='[489 chars]d30>]
E   
E   First differing element 0:
E   <Result object(id=0xf33dc5cf898c48b4b7114cffd86e8172, origin='[489 chars]a438>
E   <Result object(id=0xf211aef4abb24ecb8cc36beebbd848a8, origin='[488 chars]2d30>
E   
E   Diff is 2240 characters long. Set self.maxDiff to None to see it.
```
CC @Makman2